static/js/components/widgets/ListWrapper.js

```diff
@@ -0,0 +1,118 @@
+import React, { Component } from "react"
+import Octicon from "react-component-octicons"
+
+export default class ListWrapper extends Component {
+  /**
+   * _defaultListWrapper is a basic wrapper for the widget-list component. It tracks when the user is adding or
+   * editing a widget and passes that status down to the widget and form wrappers so that they can render appropriately
+   *
+   * Props:
+   *    renderNewWidgetForm(props): renders a New Widget Form and passes props to the FormWrapper
+   *    renderEditWidgetForm(widgetId, props): renders an Edit Widget Form for the widget with id widgetId and
+   *      passes props to the FormWrapper
+   *    renderList(props): renders the list of widgets, passing props to each widgetWrapper
+   */
+  state = {
+    editMode:       false,
+    addWidgetForm:  false,
+    editWidgetForm: null
+  }
+
+  addWidget = () => {
+    /**
+     * addWidget is the onClick for the addWidgetButton. It sets the state so that a new widget form is rendered
+     */
+    this.setState({
+      editMode:       true,
+      addWidgetForm:  true,
+      editWidgetForm: null
+    })
+  }
+
+  renderAddWidgetButton = () => {
+    /**
+     * renderAddWidgetButton creates a button to add a new widget
+     */
+    return (
+      <button
+        className="btn btn-info col add-widget-btn"
+        onClick={this.addWidget}
+      >
+        <Octicon name="plus" />
+      </button>
+    )
+  }
+
+  closeForm = () => {
+    /**
+     * closeForm sets the state so that no forms are rendered
+     */
+    this.setState({
+      addWidgetForm:  false,
+      editWidgetForm: null
+    })
+  }
+
+  toggleEditMode = () => {
+    /**
+     * toggleEditMode toggles the value of editMode
+     */
+    const { editMode } = this.state
+    this.setState({
+      editMode:       !editMode,
+      addWidgetForm:  false,
+      editWidgetForm: null
+    })
+  }
+
+  editWidget = widgetId => {
+    /**
+     * editWidget sets the state so that an edit widget form is rendered
+     *
+     * inputs:
+     *    widgetId: the widget to edit
+     */
+    this.setState({
+      editMode:       true,
+      editWidgetForm: widgetId,
+      addWidgetForm:  false
+    })
+  }
+
+  render() {
+    const { addWidgetForm, editWidgetForm, editMode } = this.state
+    const { renderNewWidgetForm, renderEditWidgetForm, renderList } = this.props
+    return (
+      <div className="bg-secondary rounded card default-list-wrapper">
+        <div
+          className="edit-widget-list-bar btn-group card-header"
+          role="group"
+        >
+          <button
+            className={`btn btn-info col edit-widget-list-btn ${
+              editMode ? "active" : ""
+            }`}
+            onClick={this.toggleEditMode}
+          >
+            <Octicon name="pencil" />
```

Comment:
Could you change this and the other usage to use material-icons classes like we do elsewhere so we don't need to pull in this extra dependency for a few icons? I think @aliceriot had advised that a few weeks ago but it got missed.